pkg/syncer/fullsync.go

```diff
@@ -148,6 +159,53 @@ func CsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer, vc s
 		return err
 	}
 
+	var vcenter *cnsvsphere.VirtualCenter
+	// Get VC instance.
+	if isMultiVCenterFssEnabled {
+		vcenter, err = cnsvsphere.GetVirtualCenterInstanceForVCenterHost(ctx, vc, true)
+		if err != nil {
+			log.Errorf("failed to get virtual center instance for VC: %s. Error: %v", vc, err)
+			return err
+		}
+	} else {
+		vcenter, err = cnsvsphere.GetVirtualCenterInstance(ctx, metadataSyncer.configInfo, false)
+		if err != nil {
+			log.Errorf("failed to get virtual center instance with error: %v", err)
+			return err
+		}
+	}
+
+	// Iterate through all the k8sPVs to find all PVs with node affinity missing and
+	// patch such PVs and their corresponding PVCs with topology discovered
+	if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
+		IsWorkloadDomainIsolationSupported {
+		var pvWithMissingNodeAffinityList [](*v1.PersistentVolume)
+		for _, pv := range k8sPVs {
+			if pv.Spec.NodeAffinity == nil {
+				pvWithMissingNodeAffinityList = append(pvWithMissingNodeAffinityList, pv)
+			}
+		}
+		if len(pvWithMissingNodeAffinityList) != 0 {
+			err := patchNodeAffinityToPVAndPVC(ctx, metadataSyncer, vcenter, pvWithMissingNodeAffinityList, pvToPVCMap)
+			if err != nil {
+				log.Errorf("FullSync for VC %s: Failed to patch node affinity for PVs. Err: %v", vc, err)
+				return err
+			}
+		}
+		// Add annotation to all PVCs, if missed to get patched in patchNodeAffinityToPVAndPVC()
+		for _, pvc := range pvToPVCMap {
+			if pvc.ObjectMeta.Annotations[annCSIvSphereVolumeAccessibleTopology] == "" {
+				err = setVolumeAccessibleTopologyForPVCs(ctx, metadataSyncer, vcenter, pvc.Namespace, pvc)
```

Comment:
We need to patch PV first and then set annotation on the PVC.
We need to do this in single full sync cycle.
Here we are calling `setVolumeAccessibleTopologyForPVCs` first and then patching PV by querying datastore URL of the volume handle.  So in the first cycle for`setVolumeAccessibleTopologyForPVCs` , we will not be able to patch PVCs.

Reply:
we are patching pv first and then pvc in  patchNodeAffinityToPVAndPVC() itself, but if patch for pv succeeds but fails for pvc, we have this additional call of setVolumeAccessibleTopologyForPVC() to patch pvc if not done already